client/src/components/profileInfoEdit/ProfileInfoEdit.js

```diff
@@ -32,17 +35,63 @@ const ProfileInfoEdit = () => {
     const { name, value } = form.target;
     setInputs({ ...profileData, [name]: value });
   };
+    let passwordSchema = yup.object().shape({
+        password: yup
+        .string()
+        .required('Please Enter your password')
+        .matches(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/,
+        "Use 8 or more characters with a mix of letters, numbers & symbols"),
+        confirmedPassword: yup
+        .string()
+        .required('Please re-enter your password')
+        .oneOf([yup.ref('password'), null], 'Passwords must match')
+    })
+  const checkPassword = async () => {
+        let valid = await passwordSchema.isValid({
+          password: {...profileData}.password,
+          confirmedPassword: {...profileData}.confirmedPassword
+      });
+        if (valid){
+            return true;
+        }else{
+            return false;
+        }
+  };
 
   const [validated, setValidated] = useState(false);
-  const handleUpdate = (event) => {
-    const form = event.currentTarget;
-    event.preventDefault();
-    if (form.checkValidity() === false) {
-      event.stopPropagation();
-    } else {
-      dispatch(updateTeam(teamId, profileData));
-    }
-    setValidated(true);
+  const handleUpdate = async (event) => {
+      const form = event.currentTarget;
+      event.preventDefault();
+      if (form.checkValidity() === false) {
+          event.stopPropagation();
+      } else {
+          var newdata;
+          if (({...profileData}.password || {...profileData}.confirmedPassword ) && ({...profileData}.password !== "" || {...profileData}.confirmedPassword !== "")) {
+              if (!checkPassword()) {
+                    setValidated(false);
+                    alert("Please enter a password at least 8 chars long, using only numbers, letters and characters");
+                    return;
+              } else {
+                  newdata = {
+                      "teamName": {...profileData}.teamName,
+                      "orgName": {...profileData}.orgName,
+                      "email": {...profileData}.email,
+                      "password": {...profileData}.password
+                  }
+                  dispatch(updateTeam(teamId, newdata));
+                  dispatch(successMessageCreator('Password Changed'));
```

Comment:
I think there's already a success toast saying team doc has been updated after the dispatch in L82, so L83 makes an extra toast show up to say password has changed. I think once you refactor the change pwd functionality to a separate modal/page you can just show one toast.